src/cpu_shader/flatten.rs

```diff
@@ -202,17 +149,68 @@ fn output_two_lines_with_transform(
     *line_ix += 2;
 }
 
-const MAX_QUADS: u32 = 16;
+fn flatten_arc(
+    path_ix: u32,
+    begin: Vec2,
+    end: Vec2,
+    center: Vec2,
+    angle: f32,
+    transform: &Transform,
+    line_ix: &mut usize,
+    lines: &mut [LineSoup],
+    bbox: &mut IntBbox,
+) {
+    let mut p0 = transform.apply(begin);
+    let mut r = begin - center;
+    let tol: f32 = 0.5;
+    let radius = tol.max((p0 - transform.apply(center)).length());
+    let x = 1. - tol / radius;
+    let theta = (2. * x * x - 1.).clamp(-1., 1.).acos();
+    const MAX_LINES: u32 = 1000;
+    let n_lines = if theta <= ROBUST_EPSILON {
+        MAX_LINES
+    } else {
+        MAX_LINES.min((std::f32::consts::TAU / theta).ceil() as u32)
+    };
```

Comment:
This formulation is different from what's in the WGSL (and the removed lines below):

```
let MIN_THETA = 0.0001;
let tol = 0.1;
let radius = max(tol, length(p0 - transform_apply(transform, center)));
let theta = max(MIN_THETA, 2. * acos(1. - tol / radius));

// Always output at least one line so that we always draw the chord.
let n_lines = max(1u, u32(ceil(angle / theta)));
```

I believe the WGSL version is correct (apart from the tolerance and MIN_THETA values, which need tuning). I would keep the CPU and WGSL versions consistent.

Reply:
I hadn't intended to change this. I'm pretty sure that what happened is that #415 didn't make it into the euler-spiral-strokes branch. I've restored it, and good catch.